app/lib/cashu/types.ts

```diff
@@ -0,0 +1,127 @@
+/**
+ * Tags are part of the data in a NUT-10 secret and hold additional data committed to
+ * and can be used for feature extensions.
+ *
+ * Tags are arrays with two or more strings being `["key", "value1", "value2", ...]`.
+ *
+Supported tags are:
+
+ * - `sigflag`: <str> determines whether outputs have to be signed as well
+ * - `n_sigs`: <int> specifies the minimum number of valid signatures expected
+ * - `pubkeys`: <hex_str> are additional public keys that can provide signatures (allows multiple entries)
+ * - `locktime`: <int> is the Unix timestamp of when the lock expires
+ * - `refund`: <hex_str> are optional refund public keys that can exclusively spend after locktime (allows multiple entries)
+ *
+ * @example
+ * ```typescript
+ * const tag: NUT10SecretTag = ["sigflag", "SIG_INPUTS"];
+ * ```
+ */
+export type NUT10SecretTag = [string, ...string[]];
+
+/**
+ * CAUTION: If the mint does not support spending conditions or a specific kind
+ * of spending condition, proofs may be treated as a regular anyone-can-spend tokens.
+ * Applications need to make sure to check whether the mint supports a specific kind of
```

Comment:
I'm not sure where the best place to put this warning is, but seems like an important note to rem